src/scalib_ext/scalib/src/sasca/bp_compute.rs

```diff
@@ -218,6 +218,30 @@ impl Distribution {
         );
         return res;
     }
+
+    // Update belief with weight average of new and previous value
+    pub fn update_dampen(
+        &mut self,
+        numerator: &Distribution,
+        denominator: &Distribution,
+        alpha: f64,
+    ) {
+        let mut self_arr = self
+            .value_mut()
+            .expect("update_dampen needs full dist self");
+        let num_arr = numerator
+            .value()
+            .expect("update_dampen needs full dist num");
+        let v: f64 = self_arr.iter().sum::<f64>();
+        if let Some(denom_arr) = denominator.value() {
+            let u: f64 = num_arr.iter().sum::<f64>() / denom_arr.iter().sum::<f64>();
```

Comment:
This will not work when distributions are true 2D arrays (i.e. you have multiple distributions). The normalization factor is then different for each distribution.

Moreover, there is an error in the formula: the normalization factor should be the sum of the ratios of the numerator and denominator, not the ratio of the sums.